pkg/hooks/manager.go

```diff
@@ -196,18 +196,30 @@ func (m *Manager) OnDefineDomain(domainSpec *virtwrapApi.DomainSpec, vmi *v1.Vir
 				}
 				defer conn.Close()
 
-				client := hooksV1alpha1.NewCallbacksClient(conn)
-
 				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 				defer cancel()
-				result, err := client.OnDefineDomain(ctx, &hooksV1alpha1.OnDefineDomainParams{
-					DomainXML: domainSpecXML,
-					Vmi:       vmiJSON,
-				})
-				if err != nil {
-					return "", err
+
+				if callback.Version == hooksV1alpha1.Version {
```

Comment:
To make it easier to add new versions later on, it might be better to use a switch statement here instead of an if-else.

Like so:
```
switch callback.Version {
    case hooksV1alpha1.Version:
        ...
    case hooksV1alpha2.Version:
        ...
    default:
        return nil, fmt.Error(...)
}
```

So to add a new version you could just add a new case statement, and if the version is unknown you get a clear error stating what the problem is.

Reply:
There are no grpc definitions for OnDefineDomain in valpha1. I'm really not sure why it is being versioned this way when things are only added and not changed tbh. @phoracek may be the best person to weigh in.

Reply:
Yep, I don't believe that this concept is really working out. If every new version needs to have the same boilerplate code this will be huge and not readable..

Reply:
I agree. We will soon introduce a versioned communication channel betwen virt-handler and virt-launcher too. @slintes looks into that. He will also there use grpc itself to do the whole negotiation.